src/ngraph/runtime/cpu/cpu_backend.cpp

```diff
@@ -90,6 +90,7 @@ shared_ptr<runtime::Executable>
                                        ngraph::pass::PassConfig& pass_config,
                                        bool performance_counters_enabled)
 {
+    std::lock_guard<std::mutex> guard(m_exec_map_mutex);
```

Comment:
we only want to guard the access and modifications to m_exec_map. make_shared<CPU_Executable>(func, pass_config, performance_counters_enabled) should be outside the critical section.

Reply:
@jbobba i addressed the changes requested , can u check?